content/docs/tasks/external_auth_opa.md

```diff
@@ -0,0 +1,133 @@
+# External Authorization
+
+## Overview and limitations
+External authorization allows offloading authorization of incoming HTTP requests to a remote endpoint not necessarily running inside the proxy's context.
+Proxies supporting external authorization will, for every request, issue a permission check to the remote endpoint. Due to the potential high number of RPS a system can have, external authorization has a number of tuneable settings to allow more or less control at the expense of performance (send HTTP payload or headers only, timeout, default behaviour if authorization fails to reply, etc.).
+
+OSM allows configuring envoy's [External Authorization extension](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/ext_authz_filter) through OSM's MeshConfig.
+Authorization filter is currently applied to inmesh `inbound` and `ingress` connections affecting HTTP request made towards any service in the mesh.
+
+The following example will show how to configure external authorization with `opa-envoy-plugin`.
+Note this example will use a single, remote (over the network) endpoint to validate all traffic. This is not a suggested production deployment.
+
+Example OSM's MeshConfig:
+
+```
+...
+inboundExternalAuthorization:
+  enable: true
+  address: opa.opa.svc.cluster.local
+  port: 9191
+  statPrefix: authz_opa
+  timeout: 1s
+  failureModeAllow: false
+```
+
+## Example Walkthrough
+
+- Deploy an [opa-envoy-plugin](https://github.com/open-policy-agent/opa-envoy-plugin), you will need to back the service with network connectivity for this walkthrough - we highly suggest to read carefully `ola-envoy-plugin`'s guide first.
+```
+kubectl apply -f https://raw.githubusercontent.com/open-policy-agent/opa-envoy-plugin/main/quick_start.yaml
+```
+Once `opa-envoy-plugin` is running and reachable through the network, you can proceed to deploy OSM's Demo, follow `demo/run-osm-demo.sh`
+```
+demo/run-osm-demo.sh  # wait for all services to come up
```

Comment:
```suggestion
cd PATH_TO_DEMO
./run-osm-demo.sh 
```

Also tell them how to verify all the services are up. You can show them the sample output, or describe what it looks like.

Reply:
this is part of the https://github.com/openservicemesh/osm/tree/main/demo#how-to-run-the-osm-automated-demo. I'll point users to the demo documentation, I don't think we should be repeating steps in multiple articles